presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClientModule.java

```diff
@@ -16,19 +16,33 @@
 import com.facebook.presto.plugin.jdbc.BaseJdbcConfig;
 import com.facebook.presto.plugin.jdbc.JdbcClient;
 import com.google.inject.Binder;
-import com.google.inject.Module;
 import com.google.inject.Scopes;
+import io.airlift.configuration.AbstractConfigurationAwareModule;
 
+import java.net.URI;
+
+import static com.google.common.base.Preconditions.checkState;
 import static io.airlift.configuration.ConfigBinder.configBinder;
 
 public class MySqlClientModule
-        implements Module
+        extends AbstractConfigurationAwareModule
 {
     @Override
-    public void configure(Binder binder)
+    protected void setup(Binder binder)
     {
         binder.bind(JdbcClient.class).to(MySqlClient.class).in(Scopes.SINGLETON);
-        configBinder(binder).bindConfig(BaseJdbcConfig.class);
+        BaseJdbcConfig config = buildConfigObject(BaseJdbcConfig.class);
```

Comment:
I've noticed that MysqlDriver is already in the classpath with the `compile` scope. Check `MySqlClient` class. You can do validation in the constructor of this class by using the `com.mysql.jdbc.Driver#parseURL` and verifying that the `DBNAME` key isn't there.